addOns/client/src/test/java/org/zaproxy/addon/client/internal/ClientSideComponentUnitTest.java

```diff
@@ -0,0 +1,447 @@
+                Arguments.of(Type.COOKIES, Type.BUTTON, 1),
+                Arguments.of(null, null, 0),
+                Arguments.of(null, Type.BUTTON, -1),
+                Arguments.of(Type.COOKIES, null, 1));
+    }
+
+    @ParameterizedTest
+    @MethodSource("getTypePairsAndExpectedResult")
+    void shouldCompareTypeAsExpected(Type first, Type second, int expected) {
+        // Given - Note: Types are actually compared by label
+        ClientSideComponent one =
+                new ClientSideComponent(
+                        Map.of(),
+                        "tagName",
+                        "id",
+                        EXAMPLE_URL,
+                        EXAMPLE_URL,
+                        "text",
+                        first,
+                        "tagType",
+                        -1);
+        ClientSideComponent two =
+                new ClientSideComponent(
+                        Map.of(),
+                        "tagName",
+                        "id",
+                        EXAMPLE_URL,
+                        EXAMPLE_URL,
+                        "text",
+                        second,
+                        "tagType",
+                        -1);
+        // When
+        int actual = one.compareTo(two);
+        // Then
+        assertThat(actual, is(equalTo(expected)));
+    }
+
+    private static Stream<Arguments> getUrlPairsAndExpectedResult() {
+        return Stream.of(
+                Arguments.of(EXAMPLE_URL, EXAMPLE_URL, 0),
+                Arguments.of(EXAMPLE_URL, ZOO_URL, -21),
+                Arguments.of(ZOO_URL, EXAMPLE_URL, 21),
+                Arguments.of(null, null, 0),
+                Arguments.of(null, EXAMPLE_URL, -1),
+                Arguments.of(EXAMPLE_URL, null, 1));
+    }
+
+    @ParameterizedTest
+    @MethodSource("getUrlPairsAndExpectedResult")
+    void shouldCompareHrefsAsExpected(String first, String second, int expected) {
+        // Given
+        ClientSideComponent one =
+                new ClientSideComponent(
+                        Map.of(),
+                        "tagName",
+                        "id",
+                        EXAMPLE_URL,
+                        first,
+                        "text",
+                        Type.BUTTON,
+                        "tagType",
+                        -1);
+        ClientSideComponent two =
+                new ClientSideComponent(
+                        Map.of(),
+                        "tagName",
+                        "id",
+                        EXAMPLE_URL,
+                        second,
+                        "text",
+                        Type.BUTTON,
+                        "tagType",
+                        -1);
+        // When
+        int actual = one.compareTo(two);
+        // Then
+        assertThat(actual, is(equalTo(expected)));
+    }
+
+    @ParameterizedTest
+    @MethodSource("getUrlPairsAndExpectedResult") // Re-use, text is text
```

Comment:
Why not rename or even change the actual text in the method and remove these comments? URLs are text too and there's no extra validation for them.